community/index/src/test/java/org/neo4j/index/internal/gbptree/GBPTreeTest.java

```diff
@@ -1565,18 +1565,16 @@ public void shouldThrowIllegalStateExceptionOnCallingNextAfterClose() throws Exc
             assertTrue( seek.next() );
             seek.close();
 
-            for ( int i = 0; i < 2; i++ )
+            try
             {
-                try
-                {
-                    // when
-                    seek.next();
-                    fail( "Should have failed" );
-                }
-                catch ( IllegalStateException e )
-                {
-                    // then good
+                while ( seek.next() )
+                {   // exhaust it, sooner or later it will need to read from the cursor and then notice it's closed
```

Comment:
We should not be able to continue using a closed cursor. Even if we get results from the read ahead buffer we need to verify that the cursor is still open.
I think this test should look like it did before to verify this behaviour.

Reply:
That means an added check on the fast path, sure we can add that, but what is it that makes that absolutely necessary, mostly being nice and fail fast and all that?

Reply:
Yes, fail fast and don't hide bugs. The check should only be a simple boolean check (not even volatile) so I think it's worth it.